pkg/upload/upload.go

```diff
@@ -59,11 +59,19 @@ func NewUploader(creds aws.Credentials) Uploader {
 		o.Credentials = provider
 		o.Region = "us-west-2"
 	})
-	uploader := manager.NewUploader(client, func(u *manager.Uploader) {
-		u.LeavePartsOnError = true
-		u.Concurrency = runtime.NumCPU()
-		u.BufferProvider = manager.NewBufferedReadSeekerWriteToPool(int(DefaultUploadPartSize) * runtime.NumCPU())
-	})
+	var uploader *manager.Uploader
+        if disableBuffer { 
+       		uploader = manager.NewUploader(client, func(u *manager.Uploader) {
+                        u.LeavePartsOnError = true
+                        u.Concurrency = runtime.NumCPU()
+                }) 
+        } else { 
+		uploader = manager.NewUploader(client, func(u *manager.Uploader) {
+			u.LeavePartsOnError = true
+			u.Concurrency = runtime.NumCPU()
+			u.BufferProvider = manager.NewBufferedReadSeekerWriteToPool(int(DefaultUploadPartSize) * runtime.NumCPU())
```

Comment:
Can't this be the only conditional line?

Reply:
probably, but I'm not sure how exactly go treats "if" statements in lambdas. I'll keep it separate for now.